src/Streamer.cxx

```diff
@@ -1,129 +1,251 @@
     }
   } else {
-    LOG(3, "Topic required");
+    LOG(0, "Topic required");
+    status_["status.consumer"] = -1;
+  }
+  if (get_offset_boundaries().value()) {
+    LOG(3, "Unable to determine lower and higher offset in topic {}",
+        topic_name);
+    status_["status.metadata"] = -1;
   }
+
+  // prepare status info
+  status_.emplace("status.size", 0);
+  status_.emplace("status.n_messages", 0);
+  status_.emplace("status.n_sources", 0);
 }
 
 BrightnESS::FileWriter::Streamer::Streamer(const Streamer &other)
-    : _topic(other._topic), _consumer(other._consumer), _offset(other._offset),
-      _begin(other._offset), _low(other._low), _partition(other._partition) {}
+    : _consumer(other._consumer), _tp(other._tp), _offset(other._offset),
+      _begin(other._offset), _low(other._low), status_(other.status_) {}
 
 BrightnESS::FileWriter::ErrorCode
 BrightnESS::FileWriter::Streamer::closeStream() {
-  auto status = _consumer->stop(_topic, _partition.value());
-  delete _topic;
-  delete _consumer;
+  BrightnESS::FileWriter::ErrorCode status;
+
+  _tp.clear();
+  status_["status.topic_partition"] = 0;
+  if (_consumer) {
+    _consumer->close();
+    status_["status.consumer"] = 0;
+    delete _consumer;
+  }
   return status;
 }
 
 int BrightnESS::FileWriter::Streamer::connect(
     const std::string &topic_name, const RdKafkaOffset &offset,
     const RdKafkaPartition &partition) {
-  if (!_topic) {
-    std::string errstr;
-    std::unique_ptr<RdKafka::Conf> tconf(
-        RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC));
-    _topic = RdKafka::Topic::create(_consumer, topic_name, tconf.get(), errstr);
-    if (!_topic) {
-      LOG(1, "{}", errstr);
-      return int(RdKafka::ERR_TOPIC_EXCEPTION);
-    }
+  // if (!_topic) {
+  //   std::string errstr;
+  //   std::unique_ptr<RdKafka::Conf> tconf(
+  //       RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC));
+  //   _topic = RdKafka::Topic::create(_consumer, topic_name, tconf.get(),
+  // errstr);
+  //   if (!_topic) {
+  //     LOG(1, "{}", errstr);
+  //     return int(RdKafka::ERR_TOPIC_EXCEPTION);
+  //   }
 
-    RdKafka::ErrorCode resp =
-        _consumer->start(_topic, _partition.value(), _offset.value());
-    if (resp != RdKafka::ERR_NO_ERROR) {
-      return int(RdKafka::ERR_UNKNOWN);
-    }
-    int64_t high, low;
-    _consumer->query_watermark_offsets(topic_name, _partition.value(), &low,
-                                       &high, 1000);
-    LOG(5, "{} -> offset low : {}\t high : {}", topic_name, low, high);
-    _low = RdKafkaOffset(low);
-    if (_offset.value() == RdKafka::Topic::OFFSET_END) {
-      _offset = RdKafkaOffset(high);
-    } else {
-      if (_offset.value() == RdKafka::Topic::OFFSET_BEGINNING) {
-        _offset = _low;
-      }
-    }
-    _begin = _offset;
-  } else {
-    LOG(2, "Cannot connect to {}: streamer already connected ({})", topic_name,
-        _topic->name());
-  }
+  //   RdKafka::ErrorCode resp =
```

Comment:
This will be addresses in a future pull request, @michele-brambilla has this already removed in a feature branch.